client/src/components/GrillLightsControls.js

```diff
@@ -95,20 +120,55 @@ export default function GrillLightsControls(props) {
     setTailLength(tailLengthVal)
   }
 
-  /* not needed now, saving for potential future use
   function updateEntireState(fetchedState) {
     setAnimation(fetchedState.animation)
     setColor({
-      red: fetchedState.red,
-      green: fetchedState.green,
-      blue: fetchedState.blue
+      r: fetchedState.r,
+      g: fetchedState.g,
+      b: fetchedState.b
     })
     setSpeed(fetchedState.speed)
     setDirection(fetchedState.direction)
     setDensity(fetchedState.density)
     setTailLength(fetchedState.tailLength)
   }
-  */
+
+  // get state from server
+  useEffect(() => {    
+    // fetch initial server state
+    fetch(STATE_URL.current)
+    .then(response => {
+      if (response.ok) {
+        response.json().then(data => {
+          console.log('Success:', data);
+          if ("animation" in data && "r" in data && "g" in data && "b" in data && "speed" in data && "direction" in data && "density" in data && "tailLength" in data) {
```

Comment:
```suggestion
          if (["animation", "r", "g", "b", "speed", "direction", "density", "tailLength"].every(e => e in data)) {
```

Reply:
this is super fancy, and i will happily do it, but can you point me to something that explains what is happening here?

Reply:
nevermind i figured it out myself lol